test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -361,6 +392,16 @@ public void ParseAssignment ()
             Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE='"));
             Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE=0\n1"));
 
+            testParse("EV_DNE", "", "EV_DNE=");
+            testParse("EV_DNE", "a b c", "EV_DNE=a b c # comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a'bc allow singleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a\"bc allow doubleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test #a$bc allow dollarSign in comment");
+
+            testParse("EV_DNE", "http://www.google.com/#anchor", "EV_DNE=http://www.google.com/#anchor");
+            testParse("EV_DNE", "#anchor", "EV_DNE=#anchor");
+            testParse("EV_DNE", "", "EV_DNE=#no value just comment");
```

Comment:
This last one still fails because it is a topic for discussion: https://github.com/tonerdo/dotnet-env/pull/82#issuecomment-1853353877

Reply:
Scratch the above, per discussion across a few other threads (sorry) I am now onboard for this change